File: space/Models/Message.cs
public class Message
{
    public int Id { get; set; }
    public string Content { get; set; } = string.Empty;
    public DateTime SentAt { get; set; }

    public int SenderId { get; set; }
    public int ReceiverId { get; set; }

    public User Sender { get; set; } = null!;
    public User Receiver { get; set; } = null!;
}

File: frontend/src/types/message.ts
export interface MessageDto {
    id: number;
    senderId: number;
    receiverId: number;
    content: string;
    sentAt: string;
}

export interface Conversation {
    userId: number;
    userName: string;
    lastMessage: MessageDto;
}

export interface MessageViewProps {
  currentChat: number | null;
  currentUserName: string;
  messages: MessageDto[];
  isConnected: boolean;
  error: string | null;
  onSendMessage: (message: string) => Promise<void>;
  isDark: boolean;
}

export interface MessageContactSidebarProps {
  conversations: Conversation[];
  currentChat: number | null;
  onLoadChat: (userId: number, userName?: string) => void;
  isConnected: boolean;
  connecting: boolean;
  isDark: boolean;
}